be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -81,7 +87,13 @@ StatusOr<DriverState> PipelineDriver::process(RuntimeState* runtime_state) {
 
                 // check whether fragment is finished beforehand before push_chunk
                 if (_fragment_ctx->is_canceled()) {
-                    return _fragment_ctx->final_status().ok() ? DriverState::FINISH : DriverState::CANCELED;
+                    cancel(_fragment_ctx->runtime_state());
+                    if (source_operator()->pending_finish()) {
```

Comment:
add some comment to explain this scenario, to prevent others from modifying mistakely.